main/lsp/LSPTypechecker.cc

```diff
@@ -349,9 +352,11 @@ vector<core::FileRef> LSPTypechecker::runFastPath(LSPFileUpdates &updates, Worke
                 subset.emplace_back(ref);
             }
         }
-        // Remove any duplicate files.
-        fast_sort(subset);
-        subset.resize(std::distance(subset.begin(), std::unique(subset.begin(), subset.end())));
+        DEBUG_ONLY(auto initialSize = subset.size();
+                   // Remove any duplicate files.
+                   fast_sort(subset);
+                   subset.resize(std::distance(subset.begin(), std::unique(subset.begin(), subset.end())));
+                   ENFORCE(subset.size() == initialSize, "subset is not free of duplicates by construction"););
```

Comment:
I think we should continue sorting in non-debug builds, too; I realize that makes the logic awkward for debug builds, but determinism in resolving and typechecking below seems like a good thing.

Reply:
My previous logic was wrong (using epochs wasn't as predictable as I had once thought).

I've rewritten this to use an `UnorderedSet`.

This potentially allocates more, but I think we can compensate for it with some future changes (e.g., keeping track of "edited files" and "extra files that need retypechecking" should enable other optimizations that is a net win—see the commit messages for more).